kong/plugins/rate-limiting/policies/init.lua

```diff
@@ -84,20 +84,27 @@ return {
         return nil, err
       end
 
-      if conf.redis_password and conf.redis_password ~= "" then
+      local times, err = red:get_reused_times()
+      if err then
+        ngx_log(ngx.ERR, "failed to get connect reused times: ", err)
+        return nil, err
+      end
+      if conf.redis_password and conf.redis_password ~= "" and times == 0 then
```

Comment:
Let's move the check for `times == 0` at the beginning of this condition, like so:
```lua
if times == 0 and conf.redis_password and conf.redis_password ~= "" then

end
```

Reply:
btw, this is slightly unrelated to the fix you are trying to write (not send the authentication frame for already established connections vs. always switching to the configured database) - this could be part of another PR (or at least another commit in the same PR, for the sake of atomicity).